src/network/connection.rs

```diff
@@ -14,7 +14,7 @@ impl Protocol {
     // Currently, linux implementation doesn't use this function.
     // Without this #[cfg] clippy complains about dead code, and CI refuses
     // to pass.
-    #[cfg(target_os = "macos")]
+    #[cfg(all(target_family = "unix", not(target_os = "linux")))]
```

Comment:
This will also enable it on OSes like openbsd && netbsd && android && dragonfly. I would prefer to use #[cfg(any(target_os = "freebsd", target_os = "openbsd"))] to specify explicitly which OS we support, rather than saying "anything not linux".

Reply:
Since we are extending support to other BSD like OS, it's a good idea to add CI for them. But this could be a different PR, your call

Reply:
Should I rename the file to something else than `bsd.rs` then? As it is only systems with `lsof` that is supported.

Reply:
Sounds good!

Reply:
> Since we are extending support to other BSD like OS, it's a good idea to add CI for them. But this could be a different PR, your call

I think this should be in a seperate PR, as the only way to get good freebsd support is by using cirrus ci, wich may be a larger issue.

Reply:
Can you change this to
```
#[cfg(any(target_os = "macos", target_os = "freebsd"))]
```
as well?
Thanks!